service/storage_service.hh

```diff
@@ -283,7 +283,7 @@ private:
 
 public:
     std::chrono::milliseconds get_ring_delay();
-    enum class mode { NONE, STARTING, JOINING, BOOTSTRAP, NORMAL, LEAVING, DECOMMISSIONED, MOVING, DRAINING, DRAINED };
+    enum class mode { MAINTENANCE, NONE, STARTING, JOINING, BOOTSTRAP, NORMAL, LEAVING, DECOMMISSIONED, MOVING, DRAINING, DRAINED };
```

Comment:
Why not add it to the end? Seems dangerous to change the order, in case it leaks into IDL.

Reply:
To not break `api/storage_service.cc:` 
```cpp
ss::is_starting.set(r, [&ss](std::unique_ptr<http::request> req) {
    return ss.local().get_operation_mode().then([] (auto mode) {
        return make_ready_future<json::json_return_type>(mode <= service::storage_service::mode::STARTING);
    });
});
```
```cpp
ss::is_initialized.set(r, [&ss](std::unique_ptr<http::request> req) {
    return ss.local().get_operation_mode().then([&ss] (auto mode) {
        bool is_initialized = mode >= service::storage_service::mode::STARTING;
        if (mode == service::storage_service::mode::NORMAL) {
            is_initialized = ss.local().gossiper().is_enabled();
        }
        return make_ready_future<json::json_return_type>(is_initialized);
    });
});
```
```cpp
ss::is_joined.set(r, [&ss] (std::unique_ptr<http::request> req) {
        return ss.local().get_operation_mode().then([] (auto mode) {
            return make_ready_future<json::json_return_type>(mode >= service::storage_service::mode::JOINING);
        });
    });
```

I can move `MAINTENANCE` to the end and change `api/storage_service.cc`